File: JavaScript/isPalindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    s = s.toLowerCase().replace(/[^a-zA-Z0-9]+/g, "")
    // var ns = s.split("").reverse().join("")
    // return s === ns

    var count = 0,
        len = s.length
    if (len % 2 === 0) {
        count = len / 2
    } else {
        if (len === 1) {
            return true
        }
        count = (len - 1) / 2
    }
    for (var i = 0; i < count; i++) {
        if (s[i] !== s.slice(-i - 1)[0]) {
            return false
        }
    }
    return true

};

File: Python/isPalindrome.py
def isPalindrome(x):
    """
    :type x: int
    :rtype: bool
    """
    if x < 0:
        return False
    div = 1
    while x / div >= 10:
        div *= 10
    while x!=0:
        l = x / div
        r = x % 10
        if l != r:
            return False
        x = (x%div) / 10
        div /= 100
    return True


    
    
        
